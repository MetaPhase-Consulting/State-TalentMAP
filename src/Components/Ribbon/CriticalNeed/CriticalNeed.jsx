import PropTypes from 'prop-types';
import Ribbon from '../Ribbon';

const CriticalNeed = ({ shortName, ...props }) => {
  const text = shortName ? 'CN' : 'Critical need';
  return <Ribbon icon="exclamation" text={text} type="cn" {...props} />;
};

CriticalNeed.propTypes = {
  shortName: PropTypes.bool,
};

CriticalNeed.defaultProps = {
  shortName: false,
};

export default CriticalNeed;
