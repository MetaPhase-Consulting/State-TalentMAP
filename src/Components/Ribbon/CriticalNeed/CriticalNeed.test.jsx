import { shallow } from 'enzyme';
import CriticalNeed from './CriticalNeed';
import toJSON from 'enzyme-to-json';

describe('CriticalNeedComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<CriticalNeed shortName />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<CriticalNeed />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
