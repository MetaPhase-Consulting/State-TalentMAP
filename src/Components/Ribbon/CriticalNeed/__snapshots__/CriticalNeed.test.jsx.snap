// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriticalNeedComponent matches snapshot 1`] = `
<Ribbon
  className=""
  containerProps={Object {}}
  cutSide="left"
  icon="exclamation"
  isWide={false}
  isWideResults={false}
  showText={true}
  text="Critical need"
  type="cn"
/>
`;
